core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/ProjectImpl.java

```diff
@@ -61,4 +154,317 @@ public ProjectImpl(User user) {
+  }
+
+  @Override
+  public void setShortName(String shortName) {
+    this.shortName = shortName;
+  }
+
+  @Override
+  public long getId() {
+    return projectId;
+  }
+
+  @Override
+  public void setId(long id) {
+    this.projectId = id;
+  }
+
+  @Override
+  @Deprecated
+  public Long getProjectId() {
+    return projectId;
+  }
+
+  @Override
+  public Collection<Sample> getSamples() {
+    return samples;
+  }
+
+  @Override
+  public Collection<Study> getStudies() {
+    return studies;
+  }
+
+  @Override
+  public Collection<ProjectOverview> getOverviews() {
+    return overviews;
+  }
+
+  @Override
+  public ProjectOverview getOverviewById(Long overviewId) {
+    for (ProjectOverview p : getOverviews()) {
+      if (p.getId() == overviewId) {
+        return p;
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public void setCreationDate(Date date) {
+    this.creationDate = date;
+  }
+
+  @Override
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  @Override
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public void setAlias(String alias) {
+    this.alias = alias;
+  }
+
+  @Override
+  @Deprecated
+  public void setProjectId(Long projectId) {
+    this.projectId = projectId;
+  }
+
+  @Override
+  public void setSamples(Collection<Sample> samples) {
+    this.samples = samples;
+    Collections.sort(Lists.newArrayList(this.samples), new AliasComparator<Sample>());
+  }
+
+  @Override
+  public void addSample(Sample sample) {
+    this.samples.add(sample);
+    Collections.sort(Lists.newArrayList(this.samples), new AliasComparator<Sample>());
+  }
+
+  @Override
+  public void setStudies(Collection<Study> studies) {
+    this.studies = studies;
+    Collections.sort(Lists.newArrayList(this.studies), new AliasComparator<Study>());
+  }
+
+  @Override
+  public void setOverviews(Collection<ProjectOverview> overviews) {
+    if (overviews == null) {
+      this.overviews = new HashSet<>();
+    }
+    this.overviews = overviews;
+    for (ProjectOverview po : overviews) {
+      po.setProject(this);
+    }
+  }
+
+  @Override
+  public ProgressType getProgress() {
+    return progress;
+  }
+
+  @Override
+  public void setProgress(ProgressType progress) {
+    this.progress = progress;
+  }
+
+  @Override
+  public Date getLastUpdated() {
+    return lastUpdated;
+  }
+
+  @Override
+  public void setLastUpdated(Date lastUpdated) {
+    this.lastUpdated = lastUpdated;
+  }
+
+  @Override
+  public boolean isDeletable() {
+    return getId() != UNSAVED_ID && getSamples().isEmpty() && getStudies().isEmpty();
+  }
+
+  @Override
+  public SecurityProfile getSecurityProfile() {
+    return securityProfile;
+  }
+
+  @Override
+  public void setSecurityProfile(SecurityProfile profile) {
+    this.securityProfile = profile;
+  }
+
+  @Override
+  public void inheritPermissions(SecurableByProfile parent) throws SecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)